src/commands/functions-config-export.ts

```diff
@@ -140,41 +104,38 @@ export default new Command("functions:config:export")
     logger.debug(`Loaded function configs: ${JSON.stringify(pInfos)}`);
     logBullet(`Importing configs from projects: [${pInfos.map((p) => p.projectId).join(", ")}]`);
 
+    let attempts = 0;
     let prefix = "";
     while (true) {
-      const errMsg = configExport.hydrateEnvs(pInfos, prefix);
+      if (attempts >= MAX_ATTEMPTS) {
+        throw new FirebaseError("Exceeded max attempts to fix invalid config keys.");
+      }
 
+      const errMsg = configExport.hydrateEnvs(pInfos, prefix);
       if (errMsg.length == 0) {
         break;
       }
       prefix = await promptForPrefix(errMsg);
+      attempts += 1;
     }
 
     const header = `# Exported firebase functions:config:export command on ${new Date().toLocaleDateString()}`;
-    const tmpdir = fs.mkdtempSync(os.tmpdir() + "dotenvs");
-    const tmpFiles = configExport.writeDotenvFiles(tmpdir, header, pInfos);
-    // Create placeholder .env and .env.local file.
-    tmpFiles.push(
-      ...configExport.writeDotenvFiles(
-        tmpdir,
-        `${header}\n# .env.local file contains environment variables for the Functions Emulator.`,
-        [{ projectId: "local" }]
-      ),
-      ...configExport.writeDotenvFiles(
-        tmpdir,
-        `${header}\n# .env file contains environment variables that applies to all projects.`,
-        [{ projectId: "" }]
-      )
-    );
-    logger.debug(`Wrote temporary .env files: [${tmpFiles.join(",")}]`);
+    const filesToWrite: Record<string, string> = pInfos.reduce((acc, info) => {
+      acc[configExport.generateDotenvFilename(info)] = configExport.toDotenvFormat(
```

Comment:
nit: this isn't an expression-based lambda. Consider using temporaries to make this cleaner to read.

Reply:
Agreed - unrolled the reduce to simplify.